### PR TITLE
Implement the editVisit command to edit patient visit details for a specific date

### PR DESCRIPTION
**Pull request by yentheng0110**

### CODE REVIEW

File: src/main/java/bookbob/functions/CommandHandler.java [hunk 1]
```diff
@@ -413,6 +413,109 @@ public void edit(String input, Records records) throws IOException {
...
+        }
+
+        // Extract visit date from input command
+        int dateStart = input.indexOf("date/");
+        assert dateStart != -1 : "Please provide a valid visit date.";
+        if (dateStart == -1) {
+            System.out.println("Please provide a valid visit date.");
+            return;
+        }
+        int dateEnd = findNextFieldStart(input, dateStart + 5);
```
Comment (coraleaf0602) [inline src/main/java/bookbob/functions/CommandHandler.java:450]: A comment in general that could be discussed with the team: 

We could replace these "magic numbers" with a constant instead?

File: src/main/java/bookbob/functions/CommandHandler.java [hunk 2]
```diff
@@ -413,6 +413,109 @@ public void edit(String input, Records records) throws IOException {
...
+            int medicationEnd = findNextFieldStart(input, medicationStart + 2);
+            String medicationsInput = input.substring(medicationStart + 2, medicationEnd).trim();
+            String[] medicationsArray = medicationsInput.split(",\\s*");
+            for (String medication : medicationsArray) {
+                newMedications.add(medication.trim());
+            }
+        }
+
+        // Update visit details if new values are provided
+        if (newDate != null) {
```
Comment (coraleaf0602) [inline src/main/java/bookbob/functions/CommandHandler.java:500]: Good error handling here

File: src/main/java/bookbob/functions/FileHandler.java [hunk 3]
```diff
@@ -191,7 +191,7 @@ private static ArrayList<String> parseList(String input) {
     public static Visit parseVisitInputString(String visitString) {
         int visitStartIndex = visitString.indexOf("[") + 1;
         int visitEndIndex = visitString.lastIndexOf("]");
-        String visitDetails = visitString.substring(visitStartIndex, visitEndIndex);
+        String visitDetails = visitString.substring(visitStartIndex, visitEndIndex).trim();
```
Comment (coraleaf0602) [inline src/main/java/bookbob/functions/FileHandler.java:194]: Nice catch! Could've easily caused error without us realising

**Review by coraleaf0602 — APPROVED** [review]

File: src/main/java/bookbob/functions/CommandHandler.java (hunk 1 above)
Comment (yentheng0110) [reply src/main/java/bookbob/functions/CommandHandler.java:450]: Great idea!

File: src/main/java/bookbob/functions/CommandHandler.java (hunk 2 above)
Comment (yentheng0110) [reply src/main/java/bookbob/functions/CommandHandler.java:500]: Thanks!

File: src/main/java/bookbob/functions/FileHandler.java (hunk 3 above)
Comment (yentheng0110) [reply src/main/java/bookbob/functions/FileHandler.java:194]: Thanks!

**Pull request merged**